### PR TITLE
Fix rubygem-rspec failure, add rubygems hpricot and rake-compiler back

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Fix rubygem-rspec failure, add rubygems hpricot and rake-compiler back

###### Change Log  
- rubygem-rspec was failing due to missing lib, added it from source
- rubygem-hpricot and rubygem-rake-compiler were not able to build due to failing BR rubygem-rspec. So, adding them back to spec-extended (no change in their spec)

###### Does this affect the toolchain?  
**NO**

###### Associated issues  
- https://microsoft.visualstudio.com/OS/_workitems/edit/39129890

###### Test Methodology
- Local build

### CODE REVIEW

File: SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec [hunk 1]
```diff
@@ -1,11 +1,13 @@
+%global debug_package %{nil}
+%global gemdir %(IFS=: R=($(gem env gempath)); echo ${R[${#R[@]}-1]})
```
Comment (PawelWMS) [inline SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec:2]: Nit: we've recently added this macro to be provided by `ruby`, so this shouldn't be necessary anymore. It's fine to leave it there, most Ruby specs currently have it (we might clean them up at some point in the future), I just wanted to let you know for future reference.

File: SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec [hunk 2]
```diff
@@ -38,18 +40,24 @@ gem build %{gem_name}
 gem install -V --local --force --install-dir %{buildroot}/%{gemdir} %{gem_name}-%{version}.gem
 mkdir -p %{buildroot}%{gem_dir}
 cp -a .%{gem_dir}/* %{buildroot}%{gem_dir}/
+#add lib folder to buildroot from Source0
+cp -r lib/ %{buildroot}%{gem_instdir}/
```
Comment (PawelWMS) [inline SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec:44]: Are the contents of `lib` pre-built (came with the source) or have they been built as part of the package build? If they are pre-built, then we shouldn't be taking them.

Comment (neha170) [reply SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec:44]: lib/ has two .rb files, rspec.rb and version.rb. They're part of the source, don't get built during package build

**Review by jslobodzian — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

File: SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS-EXTENDED/rubygem-rspec/rubygem-rspec.spec:44]: Got it, I see what you mean, looks good.

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**